app/services/order_purchase_validator.rb

```diff
@@ -0,0 +1,28 @@
+class OrderPurchaseValidator
+
+  cattr_accessor(:additional_validations) { [AllDetailsOnOrderValidator.build(NotePresenceValidator)] }
```

Comment:
Can we add some comments on this class and the AllDetailsOnOrder validator clarifying that one validates the order and the other validates all order details belonging to that order against the provided order detail validator(s)? Looking at this here, I have some context from your commit messages, but I think for later viewers comments would help a lot, especially around this call.